app/src/ui/v_configuration.py

```diff
@@ -1,9 +1,9 @@
 import streamlit as st
-from snowflake.snowpark.context import get_active_session
 from first_time_setup import get_is_first_time_setup_dismissed, \
     render as render_first_time_setup
+from snowflake.snowpark import Session
 
-session = get_active_session()
+session = Session.builder.getOrCreate()
```

Comment:
does it need to be a global var? Looks like you've mostly removed the logic that relies on `session` being defined globally. Should we consider moving this under `if __name__ == "__main__"` then?

Reply:
We could, but I thought it's not necessary. I kept it in all 3 main streamlits, and removed it from all the libraries. THe 3 main streamlits are not meant to be used as libraries, otherwise they will fail.

Reply:
See below

Reply:
will refactor - I misunderstood what you meant earlier.